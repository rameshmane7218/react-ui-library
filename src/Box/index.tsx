export { Box, BoxProps } from './Box';
