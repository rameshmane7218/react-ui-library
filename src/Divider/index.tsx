export { Divider, HrProps } from './Divider';
