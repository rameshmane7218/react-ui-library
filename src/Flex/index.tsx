export { Flex, FlexProps } from './Flex';
