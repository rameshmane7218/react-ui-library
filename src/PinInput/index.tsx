export { InputField, InputFieldProps } from './InputField';
export { PinInput, PinInputProps } from './PinInput';
export { PinInputExample } from './PinInputExample';
