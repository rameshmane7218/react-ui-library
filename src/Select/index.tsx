export { Select, SelectProps } from './Select';
export { Option, OptionProps } from './Option';
