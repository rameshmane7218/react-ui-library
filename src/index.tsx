export * from './Button';
export * from './Thing';
