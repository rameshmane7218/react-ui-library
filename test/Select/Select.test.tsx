import React from 'react';
import { render } from '@testing-library/react';
import { Option, Select } from '../../src';

describe('Select Component', () => {
  it('should render without crashing', () => {
    render(<Select placeholder="choose option"/>);
  });
  it('render placeholder option', () => {
    const { container } = render(<Select placeholder="choose option"/>);
    const option = container.querySelector("option[value='']") as HTMLElement;

    expect(option).toBeInTheDocument();
    expect(option.textContent).toEqual('choose option');
  });
  it('should render all variants', () => {
    const { asFragment } = render(
      <>
        <Select placeholder="outline" variant="outline" />
        <Select placeholder="solid" variant="solid" />
        <Select placeholder="underline" variant="underline" />
        <Select placeholder="none" variant="none" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be disabled if we pass disabled to Select', () => {
    const { container } = render(
      <Select placeholder="choose option" disabled={true}>
        <Option value="1">1</Option>
        <Option value="2">2</Option>
      </Select>
    );
    const select = container.querySelector('select') as HTMLElement;
    expect(select).toBeDisabled();
  });
});
