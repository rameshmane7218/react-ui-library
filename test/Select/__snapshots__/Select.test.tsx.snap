// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component should render all variants 1`] = `
<DocumentFragment>
  <select
    class="sc-eDvSVe TRPDq"
    width="none"
  >
    <option
      class="sc-hLBbgP gpMNOI"
      disabled=""
      selected=""
      value=""
    >
      outline
    </option>
  </select>
  <select
    class="sc-eDvSVe bhpNXf"
    width="none"
  >
    <option
      class="sc-hLBbgP gpMNOI"
      disabled=""
      selected=""
      value=""
    >
      solid
    </option>
  </select>
  <select
    class="sc-eDvSVe eivTCK"
    width="none"
  >
    <option
      class="sc-hLBbgP gpMNOI"
      disabled=""
      selected=""
      value=""
    >
      underline
    </option>
  </select>
  <select
    class="sc-eDvSVe dbcdst"
    width="none"
  >
    <option
      class="sc-hLBbgP gpMNOI"
      disabled=""
      selected=""
      value=""
    >
      none
    </option>
  </select>
</DocumentFragment>
`;
